### PR TITLE
enh: migrate to TypeScript (1/?)

### PR DESCRIPTION
**Pull request by fontanierh**
This PR migrates some of our files to Typescript:
- models -> I have used a different way of declaring the models so that it works with typescript (extending the base `Model` class) and I have explicitly declared all columns on each model in the type (including foreign keys, and fields automatically added by Sequelize such as the `id` pkey and `created_at`/`updated_at` to make TS happy)
- the DB init script -> I removed the `package.json` file from the `init` directory (don't think we need it), I changed the shell script, and now the script is ran using [tsx](https://github.com/esbuild-kit/tsx)
- a bunch of endpoints (run, datasets, datasources, keys, providers)
- the `lib/swr` file
- the `lib/providers` file

I have manually checked that everything still works. I hope I didn't miss anything !

### CODE REVIEW

File: front/init/db.ts [hunk 1]
```diff
@@ -24,4 +24,11 @@ async function main() {
   process.exit(0);
 }
 
-await main();
+main()
```
Comment (fontanierh) [inline front/init/db.ts:27]: we don't have top-level await in Typescript unless we set our `compilerOptions.target` to `"es2017"`

File: front/init/init.sh [hunk 2]
```diff
@@ -1,6 +1,3 @@
 #!/bin/sh
+env $(cat .env.local) npx tsx init/db.ts
```
Comment (fontanierh) [inline front/init/init.sh:2]: cc @spolu WDYT of this ?

File: front/pages/api/apps/[user]/[sId]/runs/index.ts [hunk 3]
```diff
@@ -108,8 +122,11 @@ export default async function handler(req, res) {
           });
 
           try {
-            for await (const chunk of streamRes.body) {
+            for await (const chunk of streamChunks(streamRes.body)) {
```
Comment (fontanierh) [inline front/pages/api/apps/[user]/[sId]/runs/index.ts:125]: TS wouldn't let me iterate directly on `streamRes.body`

File: front/pages/api/apps/[user]/[sId]/runs/index.ts [hunk 4]
```diff
@@ -108,8 +122,11 @@ export default async function handler(req, res) {
...
               res.write(chunk);
+              // TODO: figure out if .flush() does anything
+              // otherwise figure out why typescript complains
+              // @ts-ignore
```
Comment (lasryaric) [inline front/pages/api/apps/[user]/[sId]/runs/index.ts:129]: Did you figure that out?

File: front/pages/api/apps/[user]/[sId]/runs/index.ts (hunk 3 above)
Comment (lasryaric) [reply front/pages/api/apps/[user]/[sId]/runs/index.ts:125]: I looked into it, I could not find a better answer.
Looks like some new api maybe? Not sure.

File: front/pages/api/apps/[user]/[sId]/runs/index.ts (hunk 4 above)
Comment (lasryaric) [reply front/pages/api/apps/[user]/[sId]/runs/index.ts:129]: I could not find any flush() method on the ts files.

File: front/pages/api/apps/[user]/[sId]/datasets/index.ts [hunk 5]
```diff
@@ -68,8 +86,8 @@ export default async function handler(req, res) {
         attributes: ["id", "name", "description"],
       });
 
-      res.status(200).json({ datasets });
-      break;
+      res.status(200).json({ datasets } as GetDatasetResponseBody);
+      return;
```
Comment (lasryaric) [inline front/pages/api/apps/[user]/[sId]/datasets/index.ts:90]: Why switching from breaking to returning here?
I don't especially care about it, I just think I saw in another file where the code after the switch/case/break was actually important. Can't recall where exactly.

**Review by lasryaric — DISMISSED** [review]
This looks awesome! 🔥
I would recommend a double check on testing all the app before deploying it. I can take care of that tomorrow.

File: front/init/init.sh (hunk 2 above)
Comment (spolu) [reply front/init/init.sh:2]: Looks great. As long as it works I’m cool with it 👍

File: front/pages/api/apps/[user]/[sId]/datasets/index.ts (hunk 5 above)
Comment (fontanierh) [reply front/pages/api/apps/[user]/[sId]/datasets/index.ts:90]: > I just think I saw in another file where the code after the switch/case/break was actually important

Actually it's why I changed to returns. When break and return are equivalent, I feel return is less ambiguous (especially when looking at a diff, where you don't necessarily see the whole story)

Comment (fontanierh) [reply front/pages/api/apps/[user]/[sId]/datasets/index.ts:90]: definitely not strong opinion here though, I am fine with using break

File: front/pages/api/apps/[user]/[sId]/runs/index.ts (hunk 4 above)
Comment (fontanierh) [reply front/pages/api/apps/[user]/[sId]/runs/index.ts:129]: It seems to work the same without from my tests

Comment (fontanierh) [reply front/pages/api/apps/[user]/[sId]/runs/index.ts]: I removed the `.flush`

**Review by spolu — APPROVED** [review]
Tested most the of the impacted code as well as API call with streaming. Everything LGTM.

Merging!

**Pull request merged**